org_fedora_hello_world/dbus_service/installation.py

```diff
@@ -0,0 +1,115 @@
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from time import sleep
+from os.path import normpath, join as joinpath
+
+from pyanaconda.modules.common.task import Task
+
+from org_fedora_hello_world.constants import HELLO_FILE_PATH
+from org_fedora_hello_world.dbus_service.hello_world_interface import \
+    HelloWorldCalculationTaskInterface
+
+import logging
+
+log = logging.getLogger(__name__)
+
+# The installation.py file of a service/module contains tasks. A task - Task subclass - performs
+# actual work such as manipulating files. Tasks run in the service's process. To invoke them,
+# Anaconda needs also interface classes. See the file *interface.py.
+#
+# There are three possible kinds of tasks:
+#   * Configuration task - reconfigures the installation environment, and runs every time the
+#     configuration of the service changes.
+#   * Calculation task - performs costly calculations, runs every time needed.
+#   * Installation task - performs the actual installation on the new system.
```

Comment:
This actually depends on the DBus method that returns the task. The task itself can be used in all these cases.